tests/ashley/machina_extensions/forum/test_forum_view.py

```diff
@@ -364,3 +369,177 @@ def test_testing_query_param(self):
+            topic=TopicFactory(forum=forum, type=Topic.TOPIC_STICKY, views_count=7),
+            subject="TOPIC STICKY ONE",
+        )
+
+        topic12 = forum.topics.get(subject__contains="12 views_count")
+        topic9 = forum.topics.get(subject__contains="9 views_count")
+        topic6 = forum.topics.get(subject__contains="6 views_count")
+        topicSticky = forum.topics.get(subject__startswith="TOPIC STICKY ONE")
+        # Confirms that the sticky topic has neither max or lowest view_count
+        self.assertTrue(
+            topic12.views_count
+            > topic9.views_count
+            > topicSticky.views_count
+            > topic6.views_count
+        )
+
+        # Orders on column view_post
+        url = f"{url_list_topic}?o=2"
+        response = self.client.get(url)
+
+        # Sticky should stay first, we compare it with the max and the min views_count
+        self.assertContentBefore(response, "TOPIC STICKY ONE", "12 views_count")
+        self.assertContentBefore(response, "TOPIC STICKY ONE", "6 views_count")
+
+        # Reverses the order and confirms sticky topic stays on top
+        url = f"{url_list_topic}?o=-2"
+        response = self.client.get(url)
+
+        # Sticky should stay first, we compare it with the max and the min views_count
+        self.assertContentBefore(response, "TOPIC STICKY ONE", "6 views_count")
+        self.assertContentBefore(response, "TOPIC STICKY ONE", "12 views_count")
+
+    def test_testing_topic_announce(self):
+        """Controls topics that are of type announcement don't have sorted options"""
+        # Creates posts for announcement topics
+        forum = ForumFactory()
+        PostFactory(topic=TopicFactory(forum=forum, type=Topic.TOPIC_ANNOUNCE))
+        PostFactory(topic=TopicFactory(forum=forum, type=Topic.TOPIC_ANNOUNCE))
+
+        user = UserFactory()
+        assign_perm("can_read_forum", user, forum)
+        self.client.force_login(user)
+
+        url = f"/forum/forum/{forum.slug}-{forum.pk}/"
+        response = self.client.get(url)
+
+        html = lxml.html.fromstring(response.content)
+        # Select the header block of the announcement block, the first block
+        announce_block = str(
+            etree.tostring(html.cssselect(".topiclist .card-header")[0])
+        )
+
+        # Controls that announce_block is about announcements and not topics
+        self.assertTrue("Announcements" in announce_block)
+        self.assertFalse("Topics" in announce_block)
+        self.assertTrue("Replies" in announce_block)
+        self.assertTrue("Views" in announce_block)
+        self.assertTrue("Last post" in announce_block)
+
+        # There's no sortable informations
+        self.assertFalse("sortable sorted" in announce_block)
+        # There's no column that has a sorting link on
+        self.assertFalse("<a href=" in announce_block)
+        # There's no toggle sorting
+        self.assertFalse("Toggle sorting" in announce_block)
```

Comment:
Using `assertIn` and `assertNotIn` would me more appropriate here.

Reply:
better this way, thanks !